src/System.Management.Automation/namespaces/ProviderBase.cs

```diff
@@ -1384,6 +1385,16 @@ public PSHost Host
             }
         }
 
+        /// <summary>
+        /// Gets the default item separator character for this provider.
+        /// </summary>
+        public virtual char ItemSeparator => Path.DirectorySeparatorChar;
+
+        /// <summary>
+        /// Gets the alternate item separator character for this provider.
+        /// </summary>
+        public virtual char AltItemSeparator => Path.AltDirectorySeparatorChar;
```

Comment:
The file system provider (and others) doesn't override these two properties, and that means on Unix platforms, both `ItemSeparator` and `AltItemSeparator` for the file system provider (and others) are forward slash `/`.
Theoretically, this probably is correct. But it's not what those providers behave today -- file system provider, env provider, function provider, alias provider, variable provider, they all accept and normalize `/` and `\` on Unix plats.

Yes, today the registry provider also accepts and normalizes `/` and `\`. But we agreed that's not right and should be corrected in future. Therefore, the override in registry provider makes senses.
If we agree to make both `ItemSeparator` and `AltItemSeparator` the forward slash `/` for other providers on Unix plats, it would imply that we are going to change those providers to only accept `/` on Unix, which I don't think is true ...

Reply:
/cc @SteveL-MSFT @JamesWTruher @mklement0 @iSazonov for input.

Reply:
Given that `\` is an acceptable file or folder name character on Unix filesystems, unless I'm mistaken, shouldn't that be fixed?

Reply:
I agree that these properties should always reflect the de-facto behavior of the providers.

Separately, I personally think that the filesystem provider _shouldn't_ accept `\` on Unix-like platforms - but that's probably a topic for #6745.
Consistent cross-platform behavior is a wonderful thing in principle, but not if it comes at the expense of making native features inaccessible / cumbersome to access. I wish we had gone with just `/` on Unix and told people to consistently use `/` in cross-platform code.

Reply:
> Given that \ is an acceptable file or folder name character on Unix filesystems, unless I'm mistaken, shouldn't that be fixed?

Well, make the file system provider only accept `/` on Unix plats would definitely a breaking change that will affect many existing scripts.
So maybe the fix should be: if the path contains `\` in it, then first try resolving the path literally (not treating `\` as a directory separator), and if nothing is found, then treat `\` as a directory separator. If we go with a fix like this, then `\` is still accepted as a item separator in file system provider, under some conditions though.

Reply:
Having / on Unix and \ on Windows is obvious that a compromise is needed. And it is already implemented with some side effects. 
It is also clear that we will not be able to get rid of all the side effects. 
It is also clear that users will use their usual value _in console_ and most likely it will correspond to their main platform (but not the platform on which they are now located).
For scripts, we need to describe in docs how to create portable paths and implement something like Test-Path -Portable and perhaps [portablepath].

I know only one way to create a portable path - Join-Path.
__Based on this and the above, I would generally forbid the path separators, so as not to provoke users to create non-portable scripts.__

Reply:
I agree with @iSazonov, I use `Join-Path` now whenever I have a file system path as it's the predictable way to have a correct path for the target system.  Perhaps as part of the FileSystemProvider v2 thinking, we should only support `/` on non-Windows platforms, but that's a separate [discussion](https://github.com/PowerShell/PowerShell/issues/6833).  For the registry provider there's already an [issue](https://github.com/PowerShell/PowerShell/issues/5536) where it can't read a path that contains a `/` which is valid for the registry.  For this PR, it might be better to default to `Path.DirectorySeparatorChar` for `AltItemSeparator` and require explicit setting to `Path.AltDirectorySeparatorChar`.

Reply:
@SteveL-MSFT: While `Join-Path` is a clean and robust solution, it is also verbose and slow. 

**It's easy to create portable paths directly: use `/` on all platforms**, which even .NET and the Windows API understand on Windows (support is inconsistent in `cmd`).

Additionally, on both platforms it is _benign_ to accidentally double `/` when building paths with string concatenation (e.g., `Get-Item /tmp//foo` works just fine).

I think preventing the direct use of path separators would be ill-advised.

I agree that it's a separate discussion, but note that, just like with the registry provider, the current filesystem provider cannot access files with names such as `a\b` on Unix, where such filenames are legal.

Reply:
Is there a `Resolve-Path -Normalize` to convert slashes? That'd be really nice... 🤔

Reply:
@vexx32 , it's what both `Resolve-Path` (resolving in terms of PS drives) and `Convert-Path`  (resolving in terms of native paths) already do, implicitly.

While `Convert-Path` invariably resolves to a _full_ (and native) path, `Resolve-Path -Relative` works with relative paths too.

The limitation is that both cmdlets currently require the target path to _exist_ - see #5730.

 Related: #6197, #8435 , #3501

Reply:
Yep I've been trying to remove that requirement, but I can't for the life of me find the code that does that in certain cases. :(

Reply:
So, what is the action to be applied from this conversation?

Reply:
Re-tagging for @PowerShell/powershell-committee review